File: src/determinism_check/determinism_check_workflow.erl
-module(determinism_check_workflow).

% elp:ignore W0012 W0040
-moduledoc false.

-export([
    execute/2
]).

-include_lib("temporal_sdk/include/workflow.hrl").

execute(_Context, [[~"echo_activity"]]) -> start_activity(echo_activity, [[]]);
execute(_Context, [[~"hello_activity"]]) -> start_activity(hello_world_activity, [[""]]);
execute(_Context, [[~"rand_uniform"]]) -> record_rand_uniform();
execute(_Context, [[~"uuid4"]]) -> record_uuid4();
execute(_Context, _Input) -> ok.

File: lib/determinism_check/determinism_check_workflow.ex
defmodule DeterminismCheck.Workflow do
  use TemporalSdk.Workflow
  @moduledoc false

  @impl true
  def execute(_context, [["echo_activity"]]), do: start_activity(:echo_activity, [[]])
  def execute(_context, [["hello_activity"]]), do: start_activity(:hello_world_activity, [[""]])
  def execute(_context, [["rand_uniform"]]), do: record_rand_uniform()
  def execute(_context, [["uuid4"]]), do: record_uuid4()
  def execute(_context, _Input), do: :ok
end
